tensorflow/core/kernels/matmul_op.cc

```diff
@@ -591,6 +591,19 @@ TF_CALL_bfloat16(REGISTER_CPU);
 TF_CALL_int32(REGISTER_CPU);
 TF_CALL_complex64(REGISTER_CPU_EIGEN);
 TF_CALL_complex128(REGISTER_CPU_EIGEN);
+
+#elif defined(INTEL_MKL) && defined(DO_NOT_USE_ML)
+
+// MKL opensource supports only float for matrix-multiplication
+TF_CALL_float(REGISTER_CPU_EIGEN);
+TF_CALL_double(REGISTER_CPU);
+TF_CALL_half(REGISTER_CPU);
+TF_CALL_bfloat16(REGISTER_CPU);
+
+TF_CALL_int32(REGISTER_CPU);
```

Comment:
Please factor out common registrations under #if defined(INTEL_MKL) and use defined(DO_NOT_USE_ML) only for the portion that differs, that is, TF_CALL_double(...).